### PR TITLE
Removing exports of 'utilities/selection' from DetailsList and MarqueeSelection and updating imports where used

### LINKED ISSUE #9661: DetailsList - "Cannot redefine property" when accessing "/lib/DetailsList" because of SelectionMode being exported at multiple places
### Environment Information 
 
- __Package version(s)__: 7.6.3

### Please provide a reproduction of the bug in a codepen:

It should fail display the failing message :
 https://codesandbox.io/s/libdetailslist-l79rv
 
#### Actual behavior:
 
After upgrading to version 7, we noticed that our `jest` (that uses babel) tests were failing when trying to access any component from `office-ui-fabric-react/lib/DetailsList` by giving the message : 
```
TypeError: Cannot redefine property: SelectionMode
        at Function.defineProperty (<anonymous>)

[...]
at node_modules/office-ui-fabric-react/lib/utilities/src/utilities/selection/index.ts:3:1
          at Array.forEach (<anonymous>)
      at Object.<anonymous> (node_modules/office-ui-fabric-react/lib/utilities/src/utilities/selection/index.ts:3:1)
      at Object.<anonymous> (node_modules/office-ui-fabric-react/lib/components/src/components/DetailsList/index.ts:1:1)
      at Object.<anonymous> (node_modules/office-ui-fabric-react/src/DetailsList.ts:1:1)
[...]
```
We can't directly reference from `office-ui-fabric-react`.

While creating the code sandbox for this issue, I noticed that the problem occurred directly so it might not be due to our jest configuration.

It's the only component that had this problem.

#### Expected behavior:

Should we still be able to access `DetailsList` through  `office-ui-fabric-react/lib/DetailsList`?

#### Priorities and help requested:
 
Requested priority: Low
 
Thank you

### PR DESCRIPTION
**Pull request by khmakoto**
#### Pull request checklist

- [x] Addresses an existing issue: Fixes #9661
- [x] Include a change request file using `$ npm run change`

#### Description of changes

While investigating issue #9661, it became clear that we had multiple exports of `utilities/selection/index` in:
- `office-ui-fabric-react/lib/DetailsList`
- `office-ui-fabric-react/lib/MarqueeSelection`
- `office-ui-fabric-react/lib/Selection`

This created an issue where trying to access any component from these paths gave an error saying that it was redefining a property. To avoid this I'm removing the exports in both `DetailsList` and `MarqueeSelection` and updating all the places where these imports were being used.

#### Focus areas to test

(optional)

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/OfficeDev/office-ui-fabric-react/pull/9750)

### CODE REVIEW

File: packages/office-ui-fabric-react/src/components/DetailsList/index.ts [hunk 1]
```diff
@@ -1,4 +1,3 @@
-export * from '../../utilities/selection/index';
```
Comment (KevinTCoughlin) [inline packages/office-ui-fabric-react/src/components/DetailsList/index.ts]: We'll want to check ODSP to see how they reference `Selection` to ensure modifying this export doesn't break AMD imports for them.

File: packages/office-ui-fabric-react/src/MarqueeSelection.ts [hunk 2]
```diff
@@ -1,3 +1,2 @@
 export * from './components/MarqueeSelection/MarqueeSelection';
 export * from './components/MarqueeSelection/MarqueeSelection.types';
-export * from './utilities/selection/index';
```
Comment (cliffkoh) [inline packages/office-ui-fabric-react/src/MarqueeSelection.ts]: Technically a breaking change.. I would tread very carefully. Get sign off from David and Micah and make sure we maximize communication + be able to justify the change.

Comment (cliffkoh) [reply packages/office-ui-fabric-react/src/MarqueeSelection.ts]: Tagging @dzearing  and @micahgodbolt

**Comment by dzearing** [comment]
I see references in onedrive to:

```tsx
import { DetailsListLayoutMode, Selection, SelectionMode, IColumn, ConstrainMode } from 'office-ui-fabric-react/lib/DetailsList';
```

This would likely break.

**Comment by khmakoto** [comment]
@dzearing I'll close this for the moment since this will break people. I'll also check if the package breakup work we're doing helps with this issue. If it does not, we need to think about a way of dealing with this without breaking people (although I can't think of any off the top of my head).

**Comment by sushruth** [comment]
Could it be possible to export everything except `SelectionMode` from `interfaces.ts` in `index.ts`? That could potentially solve the issue for many projects wanting to upgrade to 7+ while also supporting existing projects importing `SelectionMode` the way they are. 

It will definitely make the `index.ts` file non compliant with the rest of the codebase's standard but that could potentially be addressed when the breaking change has to go in, in the future.

**Pull request closed without merging**